### DISCUSSION TITLE
CLI Command to Generate Routes

### DISCUSSION THREAD

**Discussion by Abdel-Monaam-Aouini** [vercel/next.js#66608] [Ideas]
### Goals

1. Improved Developer Experience: Simplifies the process of setting up new routes, especially for developers who are new to Next.js or prefer using CLI tools.
2. Consistency: Ensures consistent directory and file structure for routes across the project.
3. Efficiency: Saves time by automating repetitive tasks involved in route creation.

### Non-Goals

1.
2.
3.

### Background

:)

### Proposal

Description
I would like to request a new CLI command for the Next.js framework that simplifies the process of generating route files and folders. The proposed command, next generate route, would take a specified route as input and automatically create the appropriate directory structure and files needed for Next.js routing.

Proposed Command
`next generate route <route-path>`

For example, running the following command:
`next generate route /posts/:postID`

Should result in the creation of the following structure:
`/pages/posts/[postID].js`

**Comment by Netail** [comment]
This gives me Angular vibes, but I feel like it takes the same amount of time to create the directories + page as typing the cli command.
Besides people have a lot of preferences. 
- Typescript/Javascript? 
- Create (s)css file with it?
- Function or arrow function?
